src/Interval.cpp

```diff
@@ -99,17 +99,13 @@ std::string Interval::serialize () const
 std::string Interval::json () const
 {
   std::stringstream out;
-  out << '{';
+  out << "{\"id\":" << id;
 
   if (is_started ())
-    out << "\"start\":\"" << start.toISO () << "\"";
+    out << ",\"start\":\"" << start.toISO () << "\"";
 
   if (is_ended ())
-  {
-    if (is_started ())
-      out << ',';
-    out << "\"end\":\"" << end.toISO () << "\"";
-  }
+    out << ",\"end\":\"" << end.toISO () << "\"";
```

Comment:
Here too you'll want to add curly braces.

Reply:
I know that there rarely is an interval without a start, but I would prefer to keep lines 109 - 110 for consitency

Reply:
As I understand lines 109-110 were adding a comma in front only if the start existed because if a start existed, the start would be the first item in the json object but if a start did not exist, the end time would be the first item in the object.

With this change, there is always a defined first item in the object: the id.

Reply:
@j6s you are definitely right here 😅 - however I found that there is the case of an empty interval which then gets serialized as `{"id":0}` instead of `{}`.
And also tests `dom.t` and `interval.t` are broken. They appear green on Travis (should be fixed with https://github.com/GothenburgBitFactory/timewarrior/commit/dcfe0d1c), but if you inspect the individual runs you see the error reports (e.g. https://travis-ci.org/GothenburgBitFactory/timewarrior/jobs/651153570)